presto-main/src/test/java/io/prestosql/sql/query/TestHaving.java

```diff
@@ -13,18 +13,58 @@
  */
 package io.prestosql.sql.query;
 
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 public class TestHaving
 {
+    private QueryAssertions assertions;
+
+    @BeforeClass
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
     @Test
     public void testImplicitGroupBy()
     {
-        try (QueryAssertions queryAssertions = new QueryAssertions()) {
-            queryAssertions
-                    .assertQuery(
-                            "SELECT 'x' FROM (VALUES 1, 1, 2) t(a) HAVING true",
-                            "VALUES 'x'");
-        }
+        assertions.assertQuery(
+                "SELECT 'x' FROM (VALUES 1, 1, 2) t(a) HAVING true",
+                "VALUES 'x'");
+    }
+
+    @Test
+    public void testCorrelation()
```

Comment:
Shouldn't this be added to `io.prestosql.sql.query.TestSubqueries`?

Reply:
Indeed. I'll move it.